### PR TITLE
Martin's Avocado API

### PR DESCRIPTION
**Pull request by Martin-Joensson**
## Netlify link
https://avocado-data.onrender.com/

### CODE REVIEW

**Review by wwenzz — COMMENTED** [review]
Great job Martin! You have implemented all we learned last week and your code is really neat 😄 
Keep up the great work 🚀

File: server.js [hunk 1]
```diff
@@ -1,13 +1,11 @@
...
 
 // If you're using one of our datasets, uncomment the appropriate import below
 // to get started!
-// import avocadoSalesData from "./data/avocado-sales.json";
-// import booksData from "./data/books.json";
-// import goldenGlobesData from "./data/golden-globes.json";
-// import netflixData from "./data/netflix-titles.json";
-// import topMusicData from "./data/top-music.json";
+import avocadoSalesData from "./data/avocado-sales.json";
+import magicItems from "./data/magic-items.json";
```
[the remark this answers is not in the document]
Comment (AntonellaMorittu) [reply server.js:8]: true! Remove the data and imports unused please :)

File: server.js [hunk 2]
```diff
@@ -21,7 +19,44 @@ app.use(express.json());
 
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const endpoints = expressListEndpoints(app);
+  res.json(endpoints);
+});
```
Comment (AntonellaMorittu) [inline server.js:24]: when you list the endpoints you can ass more info to document your API, for example you can filter the data by queries like `region` or even `lowestprice` (check the naming convention), you could add something like this:

```javascript
app.get("/", (req, res) => {
  const usageInfo = {
    description: "Get avocado sales data.",
    queryParameters: {
      region: "Filter by region (case-insensitive).",
      lowestprice:
        "Filter out all entries at a price point higher than the specified value.",
    },
  }
  const endpoints = expressListEndpoints(app);
  const avocadoEndpointIndex = endpoints.findIndex(
    (endpoint) => endpoint.path === "/avocado"
  )

  if (avocadoEndpointIndex !== -1) {
    endpoints[avocadoEndpointIndex].usageInfo = usageInfo
  }

  res.json(endpoints)
})
``` 

Have you tested this endpoint with this query param? JavaScript is case-sensitive, so `req.query.lowestprice` will look for a query parameter named "lowestprice" exactly as it's written in the URL. If you're sending the request with a query parameter named "lowestPrice" (with a capital "P"), then `req.query.lowestprice` won't be able to retrieve its value.
You can fix your code using the correct case for the query parameter:

```javascript
const priceSearch = req.query.lowestPrice;
```

**Review by AntonellaMorittu — APPROVED** [review]
Hello Martin, well done completing your first API! 👏 
Your code indeed meets the specified requirements, nice! 😎  I left some comments to improve the documentation of the API and the usage of endpoints with queries, cool that you implemented that! Keep up the great work ✨ ⭐

**Pull request closed without merging**